### ISSUE TITLE
What is the editor used in "Episode 1: The Chess Board"

### ISSUE THREAD

**Issue by Lynchnco** [haskelllive/haskelllive#23]
The editor shown in "Episode 1: The Chess Board" is awesome.
I've been using Text Mate 2 with the Haskell Bundle and it is quite good.
However, the editor used in episode 1 is the best that I've ever seen [and I've been doing this for 40 years].
On one screen it shows the following:
1. The entry of code
2. Runs the online testing tool.
3. Provides a screen to compile and run the code entered.
What is the name of this wonderful editor.
Thank you

**Comment by schell** [comment]
I'm pretty sure he's using vim.

**Comment by Lynchnco** [comment]
Good morning Carl,
...I've used VI in the past and touched VIM.
...Poor me, I didn't realize how good it really is.
...Can you recommend a reference for me.
...It looks like a free download.
...Thank you for the info - I'll definitely look into it and let you know of my result.
Happy Spring,
Patrick

Ps
...have you done anything with Haskell and Yesod?
  ----- Original Message -----

**Comment by schell** [comment]
Hi :) Vim is open source but it's easier to install from some sources rather than others. If you're on a Mac I would recommend using homebrew (http://brew.sh/) to install gvim (graphical vim). If you're on linux I would recommend your distro's package manager to install gvim or vim. For windows I'm out of suggestions as I've never owned a windows machine. After you have gvim/vim installed you'll want haskell related plugins. Firstly you may want to to take a tour of how vim works as it's quite a bit different from other editors. For that open a terminal and run `vimtutor`, which will pull up a vim demo and tutorial.

**Comment by schell** [comment]
Also - if you have an IRC client there is a lot of haskell activity on the irc.freenode.net channel #haskell. I think the author, @reinh hangs out there often.

**Comment by Lynchnco** [comment]
Good evening,

...once again, thanks for all the useful information.
...I ran my own software consulting business for 40 years.
...my Customers all used Windows, so it is what I grew up on.
...I used Unix and Linux too but mainly on pcs.
...I finally broke down and bought a iMac, since I wanted to do some research with Ruby and Haskell.
...I must say that I was delighted with it.
...the screen is amazing and the whole thing is just great - sorry I didn't buy one years ago.

...I will take your advice and install it on my iMac.
...looking forward to working with it.

...I've been trying to get Haskell up and running with Yesod but alas with no success.
...I finally found an environment that doesn't require installing Yesod with cabal.
...I gave up on installing it myself. - it just can't be that complicated.
...hopefully, this will work for me to determine if I can garner the skills to do this and if so to install 4 apps that I've been working with for a while.
...I like Ruby and Ruby on Rails but I really like Haskell.
...I found Haskell while reading the Ruby books - it was mentioned in a number of them.
...so, one day I took a look at Haskell and put Ruby aside - if I can't get Haskell running on the web, I'll go back to Ruby.

Happy Spring,
Patrick

  ----- Original Message -----

**Comment by schell** [comment]
Yes, installing large packages with cabal can be daunting. I've found that the larger the package, the less likely it is to compile. Haskell is also going through a tough time with Mac OS X. Mavericks changed the default C compiler, which GHC relies on. All in all my experience with Haskell has been that it is an amazing language with a really awesome community, but it is lacking as a development platform. I think this will quickly change though as more tools are made. The people involved are geniuses IMO and I'm really looking forward to seeing what everyone comes up with.

**Comment by Lynchnco** [comment]
Good evening,

...I will be using an iMac OS X Mavericks.
...I will install homebrew, per your suggestion.

...I need your advice on which version of vim to install - see following:
   vim
   gvim
   macvim
...supposedly, macvim can be used grapical vim or terminal vim.
...I presume that I shouldn't use vim.
...I'm presuposed to use macvim.
...Am I correct or should I follow your guidance and install gvim?

...once homebrew is installed, I will run either: 
   brew install macvim
   or
   brew install gvim.

...hope I'm not being a pest, but I'd appreciate any advice.
...switching from Text Mate 2 to gvim will be a major effort - but I think its one worth making.

Happy Spring,
Patrick
  ----- Original Message -----

**Comment by schell** [comment]
For Mac use macvim. I use vim if I want to stay in the terminal, macvim otherwise.
